docs/rules/no-invalid-this.md

```diff
@@ -0,0 +1,294 @@
+# Disallow `this` keyword on the outside of constructors/methods (no-invalid-this)
+
+To access `this` keyword on the outside of constructors/methods is dangerous.
+Under strict mode, `this` is `undefined`, so raises an `ReferenceError`.
```

Comment:
Please teach me.
As far as I understand, `this` keyword under modules is `undefined` because below:

- [this keyword](http://www.ecma-international.org/ecma-262/6.0/#sec-this-keyword-runtime-semantics-evaluation)
- [ResolveThisBinding()](http://www.ecma-international.org/ecma-262/6.0/#sec-resolvethisbinding)  
  1. Let `envRec` be [GetThisEnvironment()](http://www.ecma-international.org/ecma-262/6.0/#sec-getthisenvironment) (returns [ModuleEnvironment](http://www.ecma-international.org/ecma-262/6.0/#sec-module-environment-records)).
    - `GetThisEnvironment()` returns an environment that its `HasThisBinding()` is `true`.
    - [ModuleEnvironmentRecord.HasThisBinding()](http://www.ecma-international.org/ecma-262/6.0/#sec-module-environment-records-hasthisbinding) returns always `true`.
  2. Return `envRec.GetThisBinding()`.
    - [ModuleEnvironmentRecord.GetThisBinding()](http://www.ecma-international.org/ecma-262/6.0/#sec-module-environment-records-getthisbinding) returns always `undefined`.